allennlp/data/tokenizers/token.py

```diff
@@ -0,0 +1,38 @@
+class Token:
+    """
+    A simple token representation, keeping track of the token's text, offset in the passage it was
+    taken from, POS tag, and dependency relation.  These fields match spacy's exactly, so we can
+    just use a spacy token for this.
+
+    Parameters
+    ----------
+    text : ``str``, optional
+        The original text represented by this token.
+    idx : ``int``, optional
+        The character offset of this token into the tokenized passage.
+    pos : ``str``, optional
+        The coarse-grained part of speech of this token.
+    tag : ``str``, optional
+        The fine-grained part of speech of this token.
+    dep : ``str``, optional
+        The dependency relation for this token.
+    text_id : ``int``, optional
+        If your tokenizer returns integers instead of strings (e.g., because you're doing byte
+        encoding, or some hash-based embedding), set this with the integer.  If this is set, we
+        will bypass the vocabulary when indexing this token, regardless of whether ``text`` is also
+        set.  You can `also` set ``text`` with the original text, if you want, so that you can
+        still use a character-level representation in addition to a hash-based word embedding.
```

Comment:
I would add a comment that this is not a field on `spacy.Token` but something extra we're augmenting it with

Reply:
Done.